smf/clickhouse/method_script.sh

```diff
@@ -13,10 +13,11 @@ DATASTORE="$(svcprop -c -p config/store "${SMF_FMRI}")"
 args=(
 "--log-file" "/var/tmp/clickhouse-server.log"
 "--errorlog-file" "/var/tmp/clickhouse-server.errlog"
+"--config-file" "/opt/oxide/clickhouse/config.xml"
```

Comment:
A fun quirk that I found when working with config files is that they seemed to not work very well with CLI flags. Can you confirm that the log files being used by ClickHouse are the ones specified here?

If not, I believe we're going to have to set everything via the config file. TBH I think I kind of prefer not mixing config files and config flags. It might get confusing over time. WDYT?

Reply:
Can confirm:

```
[ Oct  1 19:09:12 Enabled. ]
[ Oct  1 19:09:12 Rereading configuration. ]
[ Oct  1 19:09:15 Rereading configuration. ]
[ Oct  1 19:09:22 Executing start method ("/opt/oxide/lib/svc/manifest/clickhouse.sh"). ]
+ set -o errexit
+ set -o pipefail
+ . /lib/svc/share/smf_include.sh
++ SMF_EXIT_OK=0
++ SMF_EXIT_NODAEMON=94
++ SMF_EXIT_ERR_FATAL=95
++ SMF_EXIT_ERR_CONFIG=96
++ SMF_EXIT_MON_DEGRADE=97
++ SMF_EXIT_MON_OFFLINE=98
++ SMF_EXIT_ERR_NOSMF=99
++ SMF_EXIT_ERR_PERM=100
++ svcprop -c -p config/listen_addr svc:/oxide/clickhouse:default
+ LISTEN_ADDR=fd00:1122:3344:101::e
++ svcprop -c -p config/listen_port svc:/oxide/clickhouse:default
+ LISTEN_PORT=8123
++ svcprop -c -p config/store svc:/oxide/clickhouse:default
+ DATASTORE=/data
+ args=("--log-file" "/var/tmp/clickhouse-server.log" "--errorlog-file" "/var/tmp/clickhouse-server.errlog" "--config-file" "/opt/oxide/clickhouse/config.xml" "--" "--path" "${DATASTORE}" "--listen_host" "$LISTEN_ADDR" "--http_port" "$LISTEN_PORT")
+ exec /opt/oxide/clickhouse/clickhouse server --log-file /var/tmp/clickhouse-server.log --errorlog-file /var/tmp/clickhouse-server.errlog --config-file /opt/oxide/clickhouse/config.xml -- --path /data --listen_host fd00:1122:3344:101::e --http_port 8123
[ Oct  1 19:09:22 Method "start" exited with status 0. ]
Processing configuration file '/opt/oxide/clickhouse/config.xml'.
Logging trace to /var/tmp/clickhouse-server.log
Logging errors to /var/tmp/clickhouse-server.errlog
```

Reply:
> TBH I think I kind of prefer not mixing config files and config flags. It might get confusing over time

Maybe. We're already doing that, however, by mixing the _embedded config_ file and the CLI flags, and throwing in environment variables in a few places to boot. Can we file an issue to iron that out if it's important?

Reply:
Sure! An issue would be great. I can see how the values we're getting from the config file would take a little more effort to include in the config file.

Would it be ok to move the hardcoded logfile values into the config file though?

```
    <logger>
        <level>trace</level>
        <console>true</console>
        <log>/var/tmp/clickhouse-server.log</log>
        <errorlog>/var/tmp/clickhouse-server.errlog</errorlog>
    </logger>
```

It'd be nice to have all the values from that block set in the same place. WDYT?

Reply:
> Would it be ok to move the hardcoded logfile values into the config file though?

Sure, that sounds like a good idea right now. Will do.

Reply:
Done in bb3dd9909